---
title: "lux paper"
output: html_document
---

```{r read_in_file,cache=TRUE}
library(dplyr)
folder<- list.files("~/R/data")
setwd("~/R/data")
all_sample<- data.frame('name' = 1:28642)
all_sample<- tbl_dt(all_sample)
for (i in 1:length(folder)){
  temp<- read.csv(folder[i])
  temp<- tbl_df(temp)
  temp<- mutate(temp,log_reads=ifelse(Total.gene.reads==0,0,log2(Total.gene.reads)))
  all_sample[,as.character(folder[i])]<- temp$log_reads 
}
all_sample_no_name<- select(all_sample,-name)

# generate sample cluseter based on correlation based on log_Total.gene.reads.
correlation<- cor(all_sample_no_name)
distance<- as.dist(1-correlation)
par(mar=c(2,2,2,1))
plot(hclust(distance))

# heatmap for clustering based on log_Total.gene.reads.
par(mar=c(1,1,1,1))
heatmap(as.matrix(distance))
```

```{r correlation_with_log_RPKM,cache=TRUE}
library(dplyr)
folder<- list.files("~/R/data")
setwd("~/R/data")
all_sample<- data.frame('name' = 1:28642)
all_sample<- tbl_dt(all_sample)
for (i in 1:length(folder)){
  temp<- read.csv(folder[i])
  temp<- tbl_df(temp)
  temp<- mutate(temp,log_reads=ifelse(RPKM==0,0,log2(RPKM)))
  all_sample[,as.character(folder[i])]<- temp$log_reads 
}
all_sample_no_name<- select(all_sample,-name)


# generate sample cluseter based on correlation based on log_RPKM.
correlation<- cor(all_sample_no_name)
distance<- as.dist(1-correlation)
par(mar=c(2,2,2,1))
plot(hclust(distance))

# heatmap for clustering based on correlation based on log_RPKM.
par(mar=c(1,1,1,1))
heatmap(as.matrix(distance))
```

```{r gene expression number,cache=TRUE}
library(dplyr)
library(ggplot2)
library(tidyr)
folder<- list.files("~/R/data")
setwd("~/R/data")
all_sample<- data.frame('name' = 1:28642)
all_sample<- tbl_dt(all_sample)
for (i in 1:length(folder)){
  temp<- read.csv(folder[i])
  temp<- tbl_df(temp)
  all_sample[,as.character(folder[i])]<- temp$RPKM 
}

all_sample<- select(all_sample,-name)
a<- summarise_each(all_sample,funs(sum(.>0.3&.<1)))
b<- summarise_each(all_sample,funs(sum(.>1&.<10)))
c<- summarise_each(all_sample,funs(sum(.>10&.<100)))
d<- summarise_each(all_sample,funs(sum(.>100)))
all<- rbind(a,b,c,d)
all<-as.data.frame(all)
row.names(all)<-c('0.3<RPKM<1','1<RPKM<10','10<RPKM<100','RPKM>100')
all<- as.data.frame(t(all))
all<- mutate(all,sample_name=row.names(all))
all<- gather(all,'range','count',1:4)
p<- ggplot(data = all,aes(x=sample_name,y=count/1000, fill=range))+geom_bar(stat = "identity")+theme(axis.text.x=element_text(angle=90, hjust=1))+ylab('Gene number (x1000)')+xlab('Sample names')
p
```


```{r venn_diagram,cache=TRUE}
library(dplyr)
library(gplots)
folder<- list.files("~/R/venn")
setwd("~/R/venn")
all_sample<-list()
list1<- list()
list2<- list()
i=1
while (i <= length(folder)){
  temp<- read.csv(folder[i])
  temp<- tbl_df(temp)
  temp<- filter(temp,RPKM=RPKM>0.3)
  all_sample<- append(all_sample,list(temp$Name))
  if(i%%3==0&i<=12){
    over_lap<- Reduce(intersect,all_sample)
    print(class(over_lap))
    print (length(over_lap))
    name<- paste(substr(folder[i],1,nchar(folder[i])-6),'')
    print (name)
    list1[name]<- c(list(over_lap))
    all_sample<-list()
  }
  if(i%%3==0&i>12){
    over_lap<- Reduce(intersect,all_sample)
    print(class(over_lap))
    print (length(over_lap))
    name<- paste(substr(folder[i],1,nchar(folder[i])-6),'')
    print (name)
    list2[name]<- c(list(over_lap))
    all_sample<-list()
  }
  i=i+1
}
venn(list1)
venn(list2)
```